sdk/iot/Azure.Iot.Hub.Service/src/Authentication/IotHubConnectionString.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+
+namespace Azure.Iot.Hub.Service.Authentication
+{
+    internal class IotHubConnectionString
+    {
+        public const string HostNameIdentifier = "HostName";
+        public const string SharedAccessKeyIdentifier = "SharedAccessKey";
+        public const string SharedAccessKeyNameIdentifier = "SharedAccessKeyName";
+        public const string SharedAccessSignatureIdentifier = "SharedAccessSignature";
+
+        private readonly ConnectionString _connectionString;
+        private readonly string _sharedAccessPolicy;
+        private readonly string _sharedAccessKey;
+        private readonly string _sharedAccessSignature;
+
+        public IotHubConnectionString(string connectionString)
+        {
+            _connectionString = ConnectionString.Parse(connectionString);
+            _sharedAccessKey = _connectionString.GetNonRequired(SharedAccessKeyIdentifier);
+            _sharedAccessPolicy = _connectionString.GetNonRequired(SharedAccessKeyNameIdentifier);
+            _sharedAccessSignature = _connectionString.GetNonRequired(SharedAccessSignatureIdentifier);
+
+            if (!ValidateInput(_sharedAccessPolicy, _sharedAccessKey, _sharedAccessSignature))
+            {
+                throw new ArgumentException("Specify either both the sharedAccessKey and sharedAccessKeyName, or only sharedAccessSignature");
+            }
+
+            HostName = _connectionString.GetRequired(HostNameIdentifier);
+        }
+
+        public string HostName { get; }
```

Comment:
Why is this in the middle of the class?

Reply:
Is that the format? I've seen classes with parameters defined below the constructor.

Reply:
I have also seen multiple formats but usually we have private fields, public fields/properties, constructors , public methods and private methods at the very bottom.

Reply:
The msdn doc for properties is placing it below the constructor, so i'll leave it as is: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties

Reply:
I see private properties above constructor in the link you shared, public ones below constructors. Does that then convey all public properties to be below constructor?

Reply:
do you mean private properties, or the private backing fields of the public properties?